### PR TITLE
feat: #id 19350 Maximize/Restore on double click

### PR DESCRIPTION
**Pull request by amaroks**
feat: #id [19350]

[Kanban link](https://chartiq.kanbanize.com/ctrl_board/18/cards/19350/details)

**Description of change**
* Add a double click handler

**Description of testing**
1. Checkout this branch and run `npm run dev`
1. Spawn a welcome component and double click on titlebar
1. [x] Window maximizes
1. Double click again
1. [x] Window restores

### CODE REVIEW

File: src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx [hunk 1]
```diff
@@ -187,7 +188,14 @@ class WindowTitleBar extends React.Component {
 	onDropHandler() {
 		FSBL.Clients.WindowClient.cancelTilingOrTabbing({});
 	}
-
+	/**
+	 * Called when user double clicks on drag handle
+	 */
+	onDoubleClick() {
+		// Looks like Actions.clickMaximize checks the window state
+		// and toggles between maximize and minimize.
```
Comment (d4hines) [inline src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx]: What do you mean by "Looks like"? Is that how it works or not?

**Comment by d4hines** [comment]
Brad said the reason this is broken is that we used to delegate all this to the OS, and now we're taking it over via DOM.

File: src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx [hunk 2]
```diff
@@ -187,7 +188,14 @@ class WindowTitleBar extends React.Component {
 	onDropHandler() {
 		FSBL.Clients.WindowClient.cancelTilingOrTabbing({});
 	}
-
+	/**
+	 * Called when user double clicks on drag handle
+	 */
+	onDoubleClick() {
+		// Looks like Actions.clickMaximize checks the window state
```
Comment (d4hines) [inline src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx]:
```suggestion
		// Actions.clickMaximize checks the window state
```

File: src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx (hunk 1 above)
Comment (amaroks) [reply src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx]: Alright, removed `Looks like`

**Pull request merged**